### PR TITLE
Slight refactor of data pipeline

### PR DESCRIPTION
**Pull request by ramasesh**
All of our text datasets used the common pipeline function, and then
afterwards applied a custom filter (if specified).  Since this was
universal, I absorbed the filter into the pipeline.

This is mainly in preparation for serialized MNIST, but since the refactor is independent, thought I'd get it reviewed separately.

### CODE REVIEW

File: renn/data/datasets.py [hunk 1]
```diff
@@ -201,13 +204,36 @@ def _preprocess(d):
...
+                                  filter_fn,
+                                  data_dir=data_dir)
 
   # Pad remaining examples to the sequence length.
   dset = padded_batch(dset, batch_size, sequence_length)
 
   return dset
+
+
+def mnist(split,
```
Comment (nirum) [inline renn/data/datasets.py:219]: I assume this MNIST snippet is supposed to be part of a separate PR?

Comment (ramasesh) [reply renn/data/datasets.py:219]: Whoops, yeah.  That PR is open now
